core/frontend/src/RealityDataAccessProps.ts

```diff
@@ -0,0 +1,54 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module RealityData
+ */
+
+import { AuthorizedClientRequestContext } from "@bentley/itwin-client";
+
+/**
+ * All of the currently supported ProjectWise ContextShare reality data types
+ * @internal
+ */
+export enum RealityDataType {
```

Comment:
Why ugly all-caps enum instead of `export type RealityDataType = "RealityMesh3DTiles" | "OPC" | "Terrain3DTiles" | "OMR" | "Cesium3DTiles"`?

Reply:
The use of all caps was the original casing but looking back I would prefer Pascal-case. I view the enum as more useful than unioned constants since it would be more maintainable separating the string value and the enum name making comparisons easier.

Reply:
> making comparisons easier.

`if (type === RealityDataType.RealityMesh3DTiles)` isn't easier than `if (type === "RealityMesh3DTiles")`.

Reply:
Intellisense, auto-complete, and symbol renaming cannot parse strings like `"RealityMesh3DTiles"` but all three work with enums helping to prevent technical debt surrounding manual string changes.